src/commands/device/init.ts

```diff
@@ -178,29 +208,62 @@ export default class DeviceInitCmd extends Command {
 		return device.uuid;
 	}
 
-	async configureOsImage(path: string, uuid: string, options: FlagsDef) {
-		const configureCommand = ['os', 'configure', path, '--device', uuid];
-		if (options.config) {
-			configureCommand.push('--config', options.config);
-		} else if (options.advanced) {
-			configureCommand.push('--advanced');
-		}
+	async configureOsImage(
+		osImagePath: string,
+		device: { id: number; uuid: string; api_key: string },
+		options: FlagsDef,
+		configJson: ImgConfig | undefined,
+		logger: import('../../utils/logger'),
+	) {
+		let tmpConfigJsonPath: string | undefined;
+		const { promisify } = await import('util');
+		const rimraf = promisify(await import('rimraf'));
 
-		if (options['provisioning-key-name']) {
-			configureCommand.push(
-				'--provisioning-key-name',
-				options['provisioning-key-name'],
-			);
-		}
+		try {
+			const configureCommand = ['os', 'configure', osImagePath];
+			if (configJson != null) {
+				const { populateDeviceConfig } = await import('../../utils/config');
+				populateDeviceConfig(configJson, device, device.api_key);
 
-		if (options['provisioning-key-expiry-date']) {
-			configureCommand.push(
-				'--provisioning-key-expiry-date',
-				options['provisioning-key-expiry-date'],
-			);
-		}
+				const tmp = await import('tmp');
+				const tmpNameAsync = promisify(tmp.tmpName);
+				tmp.setGracefulCleanup();
 
-		await runCommand(configureCommand);
+				tmpConfigJsonPath = (await tmpNameAsync()) as string;
+				const fs = await import('fs/promises');
+				await fs.writeFile(tmpConfigJsonPath, JSON.stringify(configJson));
```

Comment:
Considering we are moving to node 24 in this PR, what do you think of moving this to https://nodejs.org/api/fs.html#fspromisesmkdtempdisposableprefix-options ?

Reply:
I would prefer to keep it as a follow-up tbh b/c the longer the changelog of the major is, the scarier and harder it becomes to process for users.

Reply:
I couldn't find how to convert `tmp.setGracefulCleanup()` to the native disposers :(

Reply:
Fair enough, just wanted us to be aware that this is possible now and we want to use it everywhere when possible